nucliadb/nucliadb/ingest/service/writer.py

```diff
@@ -525,6 +528,74 @@ async def DelWidgets(self, request: DetWidgetsRequest, context=None) -> OpStatus
             response.status = OpStatusWriter.Status.NOTFOUND
         return response
 
+    async def GetSynonyms(  # type: ignore
+        self, request: KnowledgeBoxID, context=None
+    ) -> GetSynonymsResponse:
+        kbid = request
+        response = GetSynonymsResponse()
+        txn: Transaction
+        async with self.proc.driver.managed_transaction() as txn:
+            kbobj = await self.proc.get_kb_obj(txn, kbid)
+            if kbobj is None:
+                response.status.status = OpStatusWriter.Status.NOTFOUND
+                return response
+            try:
+                await kbobj.get_synonyms(response.synonyms)
+                response.status.status = OpStatusWriter.Status.OK
+                return response
+            except Exception as e:
+                if SENTRY:
+                    capture_exception(e)
+                logger.exception(f"Errors getting synonyms")
+                response.status.status = OpStatusWriter.Status.ERROR
+                return response
+
+    async def SetSynonyms(  # type: ignore
+        self, request: SetSynonymsRequest, context=None
+    ) -> OpStatusWriter:
+        kbid = request.kbid
+        response = OpStatusWriter()
+        txn: Transaction
+        async with self.proc.driver.managed_transaction() as txn:
+            kbobj = await self.proc.get_kb_obj(txn, kbid)
+            if kbobj is None:
+                response.status = OpStatusWriter.Status.NOTFOUND
+                return response
+            try:
+                await kbobj.set_synonyms(request.synonyms)
+                await txn.commit(resource=False)
+                response.status = OpStatusWriter.Status.OK
+                return response
+            except Exception as e:
+                if SENTRY:
+                    capture_exception(e)
+                logger.exception(f"Errors setting synonyms")
+                response.status = OpStatusWriter.Status.ERROR
+                return response
+
+    async def DeleteSynonyms(  # type: ignore
```

Comment:
Consider using `DelSynonyms` following the pattern of other functions inside the class